src/cpu/flexcpu/simple_dataflow_thread.cc

```diff
@@ -132,76 +161,153 @@ SDCPUThread::advanceInst(TheISA::PCState next_pc)
 
     const InstSeqNum seq_num = lastCommittedInstNum + inflightInsts.size() + 1;
 
-    DPRINTF(SDCPUThreadEvent, "advanceInst(seq %d, pc %#x, upc %#x)\n",
+    DPRINTF(SDCPUThreadEvent, "advanceInst(seq %d, pc %#x, upc %d)\n",
                               seq_num,
                               next_pc.pc(),
                               next_pc.upc());
-    DPRINTF(SDCPUThreadEvent, "Buffer size increased to %d\n",
-                              inflightInsts.size());
+
+    StaticInstPtr static_inst = nullptr;
 
     if (isRomMicroPC(next_pc.microPC())) {
```

Comment:
This function has gotten really really long with all the different things joined into it hmm... I think some of this could be moved into a different utility/helper function like decodeInst() for code organization if for nothing else.

Reply:
Basically a named step that generates a StaticInstPtr.

Basically we have 3 steps:
  - Check if we have data (either in a macroop or in the fetch buffer) and fetch if we don't <- kinda advanceInst before
  - Decode if we do (from the macroop or the fetch buffer) <- kinda onInstDataFetched before
  - Setup the InflightInst state like dependencies <- already a function

Then a call to the next step in the pipeline (execute if ready)

Reply:
It also kind of bothers me that advanceInst/issueInst aren't a one-to-one relationship with InflightInst anymore in this implementation. Basically every other event was always one-to-one, except for the data fetch loop before, since each function represented an explicit step taken towards committing a specific instruction (hence each function taking an instruction as an argument and operating on it).

Separating out the "named step that generates a StaticInstPtr" from advanceInst/issueInst I think would be a good way to restore that, if that's something we want to maintain.